compiler/qsc_data_structures/src/line_column.rs

```diff
@@ -124,6 +136,26 @@ impl Range {
     pub fn empty(&self) -> bool {
         self.start == self.end
     }
+
+    /// Returns true if the position is within the range.
+    #[must_use]
+    pub fn contains(&self, pos: &Position) -> bool {
+        (self.start..self.end).contains(pos)
+    }
+
+    /// Intersect `range` with this range and returns a new range or `None`
+    /// if the ranges have no overlap.
+    #[must_use]
+    pub fn intersection(&self, other: &Self) -> Option<Self> {
+        let start = self.start.max(other.start);
+        let end = self.end.min(other.end);
+
+        if start <= end {
+            Some(Self { start, end })
+        } else {
+            None
+        }
+    }
 }
```

Comment:
I like these methods, but per my first comment, I think we shouldn't need them. As much as possible we should use compiler `Span`s within the implementations of language service features.

Reply:
Yes, after your suggestion these methods are not necessary anymore inside Range.

Instead, I went ahead and added them as an implementation to Span. I will refactor the code using `utils/span_contains(...)` to use `Span::contains(...)` in a follow up PR, to not bloat this one.